File: AdventOfCode2022/Day2.cs
namespace AdventOfCode2022 {
    internal static class Day2 {
        public static void Run(string path) {
            Console.WriteLine("Day Two");
            List<string> input = FileHelper.ReadFileAsList(path);

            PartOne(input);
            PartTwo(input);
        }

        private static void PartOne(List<string> input) {
            int score = 0;

            foreach (string line in input) { 
                string[] splitted = line.Split(' ');
                string player = splitted[1];
                string opponent = splitted[0];

                score += opponent == "A" && player == "X" ? 4 
                    : opponent == "A" && player == "Y" ? 8
                    : opponent == "A" && player == "Z" ? 3
                    : opponent == "B" && player == "X" ? 1
                    : opponent == "B" && player == "Y" ? 5
                    : opponent == "B" && player == "Z" ? 9
                    : opponent == "C" && player == "X" ? 7
                    : opponent == "C" && player == "Y" ? 2
                    : opponent == "C" && player == "Z" ? 6 
                    : 0;
            }

            Console.WriteLine($"Total score: {score}");
        }

        private static void PartTwo(List<string> input) {
            int score = 0;

            foreach (string line in input) {
                string[] splitted = line.Split(' ');
                string player = splitted[1];
                string opponent = splitted[0];

                score += opponent == "A" && player == "X" ? 3
                    : opponent == "A" && player == "Y" ? 4
                    : opponent == "A" && player == "Z" ? 8
                    : opponent == "B" && player == "X" ? 1
                    : opponent == "B" && player == "Y" ? 5
                    : opponent == "B" && player == "Z" ? 9
                    : opponent == "C" && player == "X" ? 2
                    : opponent == "C" && player == "Y" ? 6
                    : opponent == "C" && player == "Z" ? 7
                    : 0;
            }

            Console.WriteLine($"Total score: {score}");
        }
    }
}

File: AdventOfCodeFsharp/Day2.fs
module Day2

open System.IO
open System

let partOne opponent player score = 
    match opponent, player with
    | "A", "X" -> score + 4
    | "A", "Y" -> score + 8
    | "A", "Z" -> score + 3
    | "B", "X" -> score + 1
    | "B", "Y" -> score + 5
    | "B", "Z" -> score + 9
    | "C", "X" -> score + 7
    | "C", "Y" -> score + 2
    | "C", "Z" -> score + 6
    | _ -> score

let partTwo opponent player score = 
    match opponent, player with
    | "A", "X" -> score + 3
    | "A", "Y" -> score + 4
    | "A", "Z" -> score + 8
    | "B", "X" -> score + 1
    | "B", "Y" -> score + 5
    | "B", "Z" -> score + 9
    | "C", "X" -> score + 2
    | "C", "Y" -> score + 6
    | "C", "Z" -> score + 7
    | _ -> score

let Solve(path: string) =
    printfn "Day Two"

    let input = File.ReadAllLines(path) |> Seq.map(fun x -> x.Split(" ", StringSplitOptions.RemoveEmptyEntries))
    let mutable first = 0
    let mutable second = 0

    for item in input do
        let opponent = item[0]
        let player = item[1]

        first <- partOne opponent player first
        second <- partTwo opponent player second

    printfn "%i" first
    printfn "%i" second